doc/guides/porting/board_porting.rst

```diff
@@ -120,6 +120,14 @@ Board
 A board implements an SoC with all its features, together with peripherals
 available on the board that differentiates the board with additional interfaces
 and features not available in the SoC.
+A board port on Zephyr is typically made of two parts:
+
+- A hardware description (usually done by device tree), which specifies embedded
+  SoC reference, connectors and any other hardware components sch as LEDs,
+  buttons, sensors or communication peripheral (USB, BLE controller, ..)
+
+- A software configuration (done using Kconfig) which configures and optionally
+  enables software features that are needed to operate the board.
```

Comment:
software features are not necessarily needed to "operate the board", maybe "needed by the application"?

Reply:
Point was to highlight that features to be enabled are the minimal set required by all applications. But reading it again, agree this is not clear
What about the following ?
```
A software configuration (done using Kconfig) of features and peripheral drivers. This configuration should enable the minimal set common to every application to operate the board.
```

Reply:
It's not really the minimal @erwango , e.g. we enable ARM MPU in almost all arm boards, but that is not needed for the boards to work, neither by several applications. I think we just select a default set of components.

"A boot time configuration (via Kconfig) which selects and configures the behavior of 
software components (kernel, peripheral drivers, subsystems etc.) enabled by default on this board."

Reply:
> It's not really the minimal @erwango [...] I think we just select a default set of components.

Well, initial point of "default board configuration guidelines" was actually to define what is application responsibility and what could be directly enabled on boards (which, as a consequence, is the minimal set required by all applications).
So, this is meant to actually give guidelines here and not only documenting what is a board configuration.

> e.g. we enable ARM MPU in almost all arm boards, but that is not needed for the boards to work, neither by several applications.

I indeed hesitated in giving directions here. Ideally we should. Do we consider this should be part of default configuration (either enabled or disabled)?
(If we decide there should be a direction, this should be done in another PR as we are only clarifying the existing here)

Reply:
@ioannisg, I've pushed fixes for all comments done including proposal on this one.
Please confirm if you agree on my point above or if point is still open.